src/common/regions/index.ts

```diff
@@ -0,0 +1,176 @@
+import urlJoin from 'url-join';
+import US_STATE_DATASET from 'components/MapSelectors/datasets/us_states_dataset_01_02_2020.json';
+import countyAdjacencyMsa from 'common/data/county_adjacency_msa.json';
+import { getAbbreviatedCounty } from 'common/utils/compare';
+import RegionDB from 'common/regions/region_db';
+
+const { state_dataset, state_county_map_dataset } = US_STATE_DATASET;
+
+export type FipsCode = string;
+
+export enum RegionType {
+  COUNTY = 'county',
+  STATE = 'state',
+  CBSA = 'CBSA',
+}
+
+export abstract class Region {
+  constructor(
+    public readonly name: string,
+    public readonly urlSegment: string,
+    public readonly fipsCode: FipsCode,
+    public readonly population: number,
+    public readonly regionType: RegionType,
+  ) {}
+
+  abstract get fullName(): string;
+  abstract get abbreviation(): string;
+  abstract get relativeUrl(): string;
+
+  get canonicalUrl() {
+    return urlJoin('https://covidactnow.org', this.relativeUrl);
+  }
+}
+
+export class State extends Region {
+  constructor(
+    name: string,
+    urlSegment: string,
+    fipsCode: FipsCode,
+    population: number,
+    public readonly stateCode: string,
+  ) {
+    super(name, urlSegment, fipsCode, population, RegionType.STATE);
+  }
+
+  /**
+   * State: Washington
+   * County: King County, Washington
+   */
+  get fullName() {
+    return this.name;
+  }
+
+  get abbreviation() {
+    return this.stateCode;
+  }
+  get relativeUrl() {
+    return `us/${this.urlSegment}`;
+  }
+}
+
+export class County extends Region {
+  constructor(
+    name: string,
+    urlSegment: string,
+    fipsCode: FipsCode,
+    population: number,
+    public readonly state: State,
+    public readonly cityNames: string[],
+    private readonly adjacentCountiesFips: FipsCode[],
+  ) {
+    super(name, urlSegment, fipsCode, population, RegionType.COUNTY);
+  }
+
+  get fullName() {
+    return `${this.name}, ${this.state.name}`;
+  }
+
+  get relativeUrl() {
+    return urlJoin(this.state.relativeUrl, `county/${this.urlSegment}`);
+  }
+
+  get abbreviation() {
+    return getAbbreviatedCounty(this.name);
+  }
+  get stateCode() {
+    return this.state.stateCode;
+  }
+}
+
+// getStateName and getStateCode are helper functions that make migrating
+// some of the existing state based logic over.  Ideally we will be able
+// to remove these at some point, but they are helpful in the meantime.
+export const getStateName = (region: Region): string | null => {
+  if (region.regionType === RegionType.COUNTY) {
+    return (region as County).state.fullName;
+  }
+  if (region.regionType === RegionType.STATE) {
+    return (region as State).fullName;
+  }
+  return null;
+};
+
+export const getStateCode = (region: Region): string | null => {
+  if (region.regionType === RegionType.COUNTY) {
+    return (region as County).state.stateCode;
+  }
+  if (region.regionType === RegionType.STATE) {
+    return (region as State).stateCode;
+  }
+  return null;
+};
```

Comment:
I don't think we should implement the two functions above as they won't work for MSAs, and will generate more work for us as we refactor. Having state code and state name separated might also hide useful abstractions that we might need.

For example, if we need `King County, Washington` (`fullName`), `King County, WA` (`shortName`?) and `King Co. WA.` (`abbreviation`), using state name and code will hide the need for `shortName`.